openstf/model/serializer.py

```diff
@@ -140,8 +140,12 @@ def save_model(
             # return the latest run
             prev_run = self._find_models(pj["id"], n=1)
             # Use [0] to only get latest run id
-            prev_run_id = str(prev_run["run_id"])
-        except LookupError:
+            if not prev_run.empty:
+                prev_run_id = prev_run["run_id"][0]
+            else:
+                self.logger.info("No previous model found in MLflow", pid=pj["id"])
+                prev_run_id = None
+        except KeyError:
```

Comment:
This seems like it does the same thing twice. Shouldn't it be either the if-statement or the try-except, instead of both?

Reply:
good one, fixed it now